q2cli/builtin/tools.py

```diff
@@ -80,40 +80,149 @@ def export_data(input_path, output_path, output_format):
+            click.echo(f"{value}\t", nl=False)
+            if description:
+                click.echo(deformat_description(description))
+            else:
+                click.echo()
+    else:
+        import textwrap
+        for value, description in descriptions.items():
+            click.secho(value, bold=True)
+            if description:
+                description = deformat_description(description)
+                tabsize = 8
+                wrapped_description = textwrap.wrap(description,
+                                                    width=72-tabsize,
+                                                    initial_indent='\t',
+                                                    subsequent_indent='\t',
+                                                    tabsize=tabsize)
+                for line in wrapped_description:
+                    click.echo(f"{line}")
+            else:
+                click.secho("\tNo description", italic=True)
+            click.echo()
+
+
+def deformat_description(description):
+    import re
+    deformatted = re.sub(r"[\t\n]+", ' ', description)
+    despaced = re.sub(r" +", ' ', deformatted)
+    return despaced
+
+
+def get_matches(words, possibilities, cutoff=0.5):
+    from difflib import get_close_matches
+    matches = []
+    for word in words:
+        matches += get_close_matches(word,
+                                     possibilities,
+                                     n=len(possibilities),
+                                     cutoff=cutoff)
+        # substring search
+        if cutoff != 1:
+            for possibility in possibilities:
+                if word.lower() in possibility.lower():
+                    if possibility not in matches:
+                        matches.append(possibility)
+
+    return matches
+
+
+@tools.command(
+        name='show-types',
+        help='List the available semantic types.',
+        short_help='',
+        cls=ToolCommand
+)
+@click.argument('types', nargs=-1)
+@click.option('--strict', is_flag=True,
+              help='Show only exact matches for the type argument(s).')
+@click.option('--tsv', is_flag=True,
+              help='Print as machine readable tab separated values.')
+def show_types(types, strict, tsv):
+    pm = q2cli.util.get_plugin_manager()
+
+    if types and strict:
+        matches = get_matches(types, list(pm.artifact_classes), 1)
+    elif types:
+        matches = get_matches(types, list(pm.artifact_classes))
+    else:
+        matches = list(pm.artifact_classes)
+
+    descriptions = {}
+    for match in sorted(matches):
+        description = pm.artifact_classes[match].description
+        descriptions[match] = description
+
+    print_descriptions(descriptions, tsv)
+
+
+@tools.command(
+        name='show-formats',
+        help='List the availabe formats.',
+        short_help='',
+        cls=ToolCommand
+)
+@click.argument('formats', nargs=-1)
```

Comment:
I think these two new functions would be a little easier to follow if `formats` and `types` were called `queries`.